presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -970,6 +970,24 @@ public void revokeTablePrivileges(Session session, QualifiedObjectName tableName
         return ImmutableList.copyOf(grantInfos.build());
     }
 
+    @Override
+    public boolean isWritingStagingFilesEnabled(Session session)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, session.getCatalog().orElseThrow(() -> new IllegalStateException("No catalog specified")));
```

Comment:
nit: I prefer to have line break to help read

```java
        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(
                session,
                session.getCatalog()
                        .orElseThrow(() -> new IllegalStateException("No catalog specified")));
```

Reply:
What if catalog was specified explicitly? 

Say

```
INSERT INTO prism.table
SELECT FROM ...
```

I would expect this method to accept `catalogName`